src/recordlinker/linking/link.py

```diff
@@ -47,18 +55,32 @@ def compare(
     # keyword arguments to pass to comparison functions and matching rule
     kwargs: dict[typing.Any, typing.Any] = algorithm_pass.kwargs
 
+    missing_field_weights = 0.0
     results: list[float] = []
     details: dict[str, typing.Any] = {"patient.reference_id": str(patient.reference_id)}
     for e in evals:
         # TODO: can we do this check earlier?
         feature = schemas.Feature.parse(e.feature)
         if feature is None:
             raise ValueError(f"Invalid comparison field: {e.feature}")
-        # Evaluate the comparison function and append the result to the list
-        result: float = e.func(record, patient, feature, **kwargs)  # type: ignore
-        results.append(result)
+        
+        # Evaluate the comparison function, track missingness, and append the 
+        # score component to the list
+        result: tuple[float, bool] = e.func(
+            record, patient, feature, missing_points_proportion, **kwargs
+        )  # type: ignore
+        if result[1]:
+            # If the field was missing, the log-odds weight was multiplied, so dividing
+            # here reverses the change without needing to overly pass the odds dict
+            missing_field_weights += result[0] / missing_points_proportion
```

Comment:
Is `missing_points_proportion` the same as `missing_field_compare_fraction` in models/algorithm.py? If they are the same, I'm worried that setting the `missing_points_proportion` to 0.0 would break here. 

Can you add a test to address this scenario?

Reply:
Should be handled now that we're passing in the log odds dict directly 👍